### DISCUSSION TITLE
BitOps Meeting (14 Sep 2022): BitOps Brainstorming session, Twitch Demo ideas, Roadmap

### DISCUSSION THREAD

**Discussion by arm4b** [bitovi/bitops#301] [Announcements]
Join the bi-weekly BitOps Community `1 hour` Meeting on `14 Sep 2022, 12:00 PM US East` to discuss the ongoing project developments, updates and plans.

See https://github.com/bitovi/bitops/discussions/288 for more info on how to join.

### Brainstorming Session: What are BitOps Benefits?
Based on feedback, it's not clear for many the benefits of BitOps as a tool. Let's brainstorm and think about what could be BitOps Benefits and Values that the project provides? This should be something easy so a non-too-technical person could understand: Project Managers, Directors, or maybe developers not familiar with DevOps and Infra as Code. Come up with ideas and explain it in simple words so we can list them on https://www.bitovi.com/bitops afterwards. 

### Follow-ups from the previous #294 Meeting
- Park BitOps Twitter/LN social accounts for the future
- BitOps Videos ideas/plans
- Update the BitOps StackStorm/HeyEmoji blogs to v2

### BitOps Demo on Twitch
Based on @dylan-bitovi idea from the previous meeting https://github.com/bitovi/bitops/discussions/294, we could use Bitovi Twitch and host something like a BitOps demo. Any ideas on what could be a good topic example for a demo, any interested folks hosting it, event format, marketing pre-req, prep work?

### Roadmap https://github.com/bitovi/bitops/discussions/245 & Development Discussions
* Create a Roadmap page (docs)?
* [BitOps Plugin support for private repos #60](https://github.com/bitovi/bitops/issues/60)
* [Parse passwords / credentials out of logging output #208](https://github.com/bitovi/bitops/issues/208)
* [Project Quality](https://github.com/bitovi/bitops/projects/4) as KPI
  * Documentation:
    * #282
    * https://github.com/bitovi/bitops/issues/143
    * https://github.com/bitovi/bitops/issues/280
    * Cleanup docs from `TODOs`, remove unimplemented features
  * Testing
    * https://github.com/bitovi/bitops/issues/286
    * https://github.com/bitovi/bitops/issues/271
    * https://github.com/bitovi/bitops/issues/287
  * CI/CD
    * https://github.com/bitovi/bitops/issues/293
    * https://github.com/bitovi/bitops/issues/224
  * Stability/bugfixes
  * Feature consistency
* Ops Repo Composability (needs a dedicated issue w description)
* [Default folder support](https://github.com/bitovi/bitops/issues/177) follow-up

**Comment by mickmcgrath13** [comment]
@armab can we add the following to the agenda?
https://github.com/bitovi/bitops/pull/257#issuecomment-1232225549

**Reply by arm4b** [reply]
Adding meetings schedule to the docs, - I guess we just want to commit that and call it done

**Comment by mickmcgrath13** [comment]
@armab 

Also, this is a relevant issue for the first agenda item:
https://github.com/bitovi/bitops/issues/143

..may be a good place to capture notes

**Reply by arm4b** [reply]
👍 Nice. Added to the Docs section in the first message

**Comment by mickmcgrath13** [comment]
Discussion for roadmap: https://github.com/bitovi/bitops/discussions/245

**Reply by arm4b** [reply]
Right, that was the source for the meeting agenda on the first Roadmap discussions 👍 
Updated the first message

**Comment by arm4b** [comment]
Attendees
@mickmcgrath13 @armab @PhillypHenning and @heatherveva.


## Meeting Minutes

### Brainstorming Session: What are BitOps Benefits?
Benefits & Values from the business & decision-making perspective
  - described in 3-4 words
  - we'll include them on the landing page bitovi.com/bitops
  - 👍 IDEA: link each point to the respective documentation
    - value (business) -> links to -> proof (technical)

We want to split the docs vs landing page target groups:
  - The landing page is more about business benefits
    - or maybe the first 30sec visible part when scrolling
    - so the decision makers (directors, managers) can understand it and see the value
    - then explore the features in more detail for tech leads, architects, etc
  - Docs are more about technical details
    - so someone adopting it would understand what are the features and how to use the tool
  - Example flow 1:
    - A tech person (not necessarily Devops) makes a decision to try the tool based on the landing page -> documentation
  - Example flow 2:
    - Decision maker (director, manager, etc) visits the landing page really quick
      - he then redirects the decision/research/PoC task to the tech person (tech leads, architects, developers)

BitOps Values & Benefits Examples:
- **BitOps keeps the working environment clean**
- **Organizes your deployment configuration** -> link to the repo pattern
- **Faster Developer Onboarding**
    - so people don’t need to understand IaC tool
      - they can start straight away
    - replace "easy" -> something else
- **Integrates with your favorite tools** -> link to the plugins
    - Use your favorite tools (Ansible, Terraform, Helm, AWS, etc)
    - plugin ecosystem
- **Saves weeks of engineering efforts** -> link to the examples
- Auditing? -> BitOps config documentation
- **Encourages GitOps best practices**
  - ALTERNATIVE: Built on GitOps best practices
  - we should mention GitOps for sure
- **Open-Source and free**
  - the project is open-source
    - meaning: it's a free tool, has documentation, could be modified/forked, good standards, etc
- **Security**
    - no storage of secrets
      - feature: auto-deletes the container with the secrets after the run
    - plugins: scan the images on Docker hub
    - custom plugins: users can own the build pipeline

Random thought from Heather:
- Downloadable PDF on the landing page?
  - to get the comprehensive BitOps details in a PDF
  - as an alternative format of sharing the information

An alternative approach to highlighting the Values on the landing page:
- Here is the Problem -> Here is the Solution that BitOps solves

### Follow-ups from the previous https://github.com/bitovi/bitops/discussions/294 Meeting
- @mickmcgrath13 and @heatherveva will explore parking the BitOps social accounts:
  - LN, Twitter, Youtube
- @PhillypHenning will do the initial short BitOps video
- @heatherveva will do the series of videos afterwards
  - create a BitOps video playlist
  - IDEA: cross-link the videos to each other
    - Youtube inline video link
- IDEA: Highlight the BitOps values & benefits (as mentioned) in one of the videos

### BitOps Demo on Twitch
- Sounds like a good thing to do
- Explore more the pre-req:
  - marketing
  - planning
  - topic
  - who wants to run it
- To follow-up in the next meeting

### Roadmap & Development Discussions
To be discussed in the next meetings

**Comment by ConnorGraham** [comment]
> schematizes cli arguments

**Comment by imaustink** [comment]
To me, BitOps is the ultimate DevOps framework. It takes care of much of the heavy lifting and lets us focus on our infrastructure needs while forcing us to be organized and modular with the way we put things together.
